test/compiler/codegen.jl

```diff
@@ -794,3 +794,6 @@ end
 f48085(@nospecialize x...) = length(x)
 @test Core.Compiler.get_compileable_sig(which(f48085, (Vararg{Any},)), Tuple{typeof(f48085), Vararg{Int}}, Core.svec()) === nothing
 @test Core.Compiler.get_compileable_sig(which(f48085, (Vararg{Any},)), Tuple{typeof(f48085), Int, Vararg{Int}}, Core.svec()) === Tuple{typeof(f48085), Any, Vararg{Any}}
+
+# Make sure that the bounds check is elided in tuple iteration
+@test !occursin("call void @", get_llvm(iterate, Tuple{NTuple{4, Float64}, Int64}))
```

Comment:
These sorts of checks usually break our coverage and debug tests (since `-g2` may add `void @llvm.dbg.value` calls, for example). It is preexisting, so does not need to block merging. But can we do any better though?

Reply:
We could maybe check for the jl_ prefix, since the boundscheck we're trying to avoid here is an intrinsic function, but I did want to test that we don't call any other non-inlined function that might be hiding the call either.